### PR TITLE
Add confirm page

### PR DESCRIPTION
**Pull request by RoshniCh**
page created with the parameter as id. form will be shown with data populated. and editable. Confirm and delete buttons dont have functionality. we added colour/font styling. merged with nav bar. tweaked some navbar colours

### CODE REVIEW

**Comment by emilypatsko** [comment]
I think the test fails because the form relies on the database to render. Not sure what to do about this!

File: ClientApp/src/NavBar/NavBar.scss [hunk 1]
```diff
@@ -21,34 +21,45 @@
...
-  color: white;
+  color: $light-text;
   text-align: center;
   padding: 14px 16px;
   text-decoration: none;
 }
   
-li a:hover:not(.active) {
-  background-color: #111;
+// li a:hover:not(.active) {
```
Comment (ChristianMcc) [inline ClientApp/src/NavBar/NavBar.scss]: not needed? :)

File: ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
+@import "../../Styles/constants";
+@import "../../Styles/styles";
+.submit-sighting-page {
+  margin: $gutter-width;
+
+  .title {
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.scss]: Is this not in styles.scss already?

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (ChristianMcc) [inline package-lock.json:1]: This file can be deleted :)

File: ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      setLatitude(response.latitude.toString());
+      setLongitude(response.longitude.toString());
+      setDescription(response.description);
+      setSightedAt(response.sightedAt);
+      setSubmittedByName(response.submittedByName);
+      setSubmittedByEmail(response.submittedByEmail);
+    });
+  }, []);
+
+  function confirmOrDeleteSighting(event: FormEvent) {
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx:38]: This isn't fully implemented of course but I think we could just have 2 separate functions here that each button calls separately instead of calling one and trying to figure out which button was pressed :)

**Review by ChristianMcc — COMMENTED** [review]
Looking good so far, couple of minor comments and yes I think you'll need to look at jest and mocking out the fetch call in your component for your test - something like this might help https://www.leighhalliday.com/mock-fetch-jest :) Alternatively the quick fix would be to take out the memory router and the async part of the test but then we can only check that it renders

File: package-lock.json (hunk 3 above)
Comment (emilypatsko) [reply package-lock.json:1]: properly deleted, or added to the .gitignore?

File: ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
+import React, { FormEvent, useState, useEffect } from "react";
+import { useParams } from "react-router";
+import { getSighting } from "../../Api/apiClient";
+import "./ConfirmSighting.scss";
+
+export function ConfirmSightingForm(): JSX.Element {
+  const [status, setStatus] = useState("LOADING");
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx]: Could probably change this to a boolean of isLoading or something instead of matching on a string here :)

File: ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx (hunk 4 above)
Comment (ChristianMcc) [reply ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.tsx:38]: Had another look as you've suggested this way should be alright, I think my personal preference is the former but I've been persuaded that the internet thinks this way is just as good/better!

File: ClientApp/package-lock.json [hunk 6]
```diff
@@ -5,17 +5,17 @@
   "requires": true,
```
Comment (ChristianMcc) [inline ClientApp/package-lock.json]: The package.json file hasn't actually changed so I'm surprised these have all updated, I don't think they should have - could we revert the changes to this file and the package.json file please?

File: ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.test.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { ConfirmSightingForm } from './ConfirmSighting';
+import ReactDOM from 'react-dom';
+import { MemoryRouter } from 'react-router-dom';
+import { scotland } from '../GettingStarted/LocationData';
+
+jest.mock("../../Api/apiClient", () => {
+  return {
+    getSighting: jest.fn().mockImplementation(() => {
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/ConfirmSighting/ConfirmSighting.test.tsx:9]: Nice, this looks good to me :)

**Review by ChristianMcc — COMMENTED** [review]
Quick comment re the package.json files and hten it's ready to merge!

**Review by ChristianMcc — APPROVED** [review]
Nice one! thanks for doing all these changes :)
